OsmAnd-java/src/main/java/net/osmand/binary/RouteDataObject.java

```diff
@@ -368,7 +368,11 @@ public String getDestinationName(String lang, boolean transliterate, boolean dir
 				destinationTagLang = "destination:lang:" + lang;
```

Comment:
The code here looks strange and probably is wrong (destination:ref needs to be ref) otherwise shield is not correctly displayed probably:
		//Issue #3289: Treat destination:ref like a destination, not like a ref